_posts/2019-99-99-vmem-split.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Vmem is split out of PMDK
+author: kilobyte
+layout: post
+identifier: vmem-split
+---
+
+### Introduction
+
+We have just split **libvmem** and its companion **libvmmalloc** out of the
+PMDK tree.  They now live in a
+[separate repository](https://github.com/pmem/vmem/), and will follow their
+own release cadence.  And, as these libraries are considered mature and
+finished, no new releases are planned once the split has been tested and
+tagged -- except for defects and new requirements of underlying platforms.
+
+### Further development
+
+**libvmem** remains the only way to use filesystem-managed persistent memory
+for volatile allocations on Windows.
+
+On Linux, though, you are better served by
+[memkind](http://memkind.github.io/memkind/) instead of **libvmem** in new
+code -- it provides extra features such as NUMA awareness and handling of
+other kinds of memory.
+
+As for **libvmmalloc**, there's no direct equivalent yet, but incoming
+kernel features will allow redirecting an unported program to volatile
```

Comment:
This is upstream in 5.2 AFAIK.

Can you confirm with Kuba?

Reply:
Right, automatic creation of fake NUMA nodes with real PMEM might work.  That it doesn't work out of the box on my frankensteinian faked setup may be my fault.  But indeed, let's ask @jschmieg.